server/eclipse-project/src/main/java/us/freeandfair/corla/auth/AbstractAuthentication.java

```diff
@@ -243,55 +291,84 @@ public boolean isAuthenticatedAs(final Request the_request,
     boolean result = false;
     final Object auth_stage_attribute =
         the_request.session().attribute(AuthenticationInterface.AUTH_STAGE);
+    final Object admin_attribute = the_request.session().attribute(ADMIN);
+    
     if (auth_stage_attribute instanceof AuthenticationStage &&
-        ((AuthenticationStage) auth_stage_attribute) == SECOND_FACTOR_AUTHENTICATED) {
-      final Object admin_attribute = the_request.session().attribute(ADMIN);
-      if (admin_attribute instanceof Administrator) {
-        final Administrator admin = (Administrator) admin_attribute;
-        result = admin.type() == the_type &&
-            the_username.equals(admin.username());
-      } else if (admin_attribute != null) {
-        // this should never happen since we control what's in the session object,
-        // but if it does, we'll clear out that attribute and thereby force another
-        // authentication
-        Main.LOGGER.error("Invalid admin type detected in session.");
-        deauthenticate(the_request, the_username);
-      }
+        admin_attribute instanceof Administrator) {
+      final AuthenticationStage stage = (AuthenticationStage) auth_stage_attribute;
```

Comment:
stage can be null here, right?  I guess it won't hurt given the below comparison to the enum?